src/Avalonia.Base/StyledElement.cs

```diff
@@ -830,59 +835,25 @@ private void ClearLogicalParent(IList children)
             }
         }
 
-        private void DetachStyles()
+        private void DetachStyles(IReadOnlyList<StyleBase>? styles = null)
```

Comment:
Based on usage here and how this list is built - maybe it makes sense to instead build a `HashSet`? This would allow for cheap lookup.

Reply:
Yeah probably, though this method comes from the code before this PR (i.e. there was already a `DetachStyles(IReadOnlyList<StyleBase>? styles)` method). Detaching styles will be refactored in future anyway because after this PR we should be able to remove individual frames.

I'd rather leave it as-is in this PR as it's not really related to the value store.